src/examples/graph.js

```diff
@@ -529,31 +555,47 @@ class Graph extends React.Component<IGraphProps, IGraphState> {
       const newIDs = {};
 
       // Every node position is relative to the top and left-most corner
-      const newNodes = copiedNodes.map(copiedNode => {
-        const x = mouseX + (copiedNode.x - cornerX);
-        const y = mouseY + (copiedNode.y - cornerY);
-        const id = `${copiedNode.id}_${Date.now()}`;
-
-        newIDs[copiedNode.id] = id;
-
-        return {
-          ...copiedNode,
-          id,
-          x,
-          y,
-        };
-      });
-
-      const newEdges = copiedEdges.map(copiedEdge => {
-        return {
-          ...copiedEdge,
-          source: newIDs[copiedEdge.source],
-          target: newIDs[copiedEdge.target],
-        };
-      });
+      const newNodes = new Map(
+        Array.from(copiedNodes.values()).map(copiedNode => {
+          const x = mouseX + (copiedNode.x - cornerX);
+          const y = mouseY + (copiedNode.y - cornerY);
+          const id = `${copiedNode.id}_${Date.now()}`;
+
+          newIDs[copiedNode.id] = id;
+
+          return [
+            id,
+            {
+              ...copiedNode,
+              id,
+              x,
+              y,
+            },
+          ];
+        })
+      );
 
-      graph.nodes = [...graph.nodes, ...newNodes];
-      graph.edges = [...graph.edges, ...newEdges];
+      const newEdges =
+        copiedEdges != null
+          ? new Map(
+              Array.from(copiedEdges.values()).map(copiedEdge => {
+                const source = newIDs[copiedEdge.source];
+                const target = newIDs[copiedEdge.target];
+
+                return [
+                  `${source}_${target}`,
+                  {
+                    ...copiedEdge,
+                    source,
+                    target,
+                  },
+                ];
+              })
+            )
+          : new Map();
+
+      graph.nodes = [...graph.nodes, ...Array.from(newNodes.values())];
```

Comment:
Map `.values()` returns an iterator, so you should be able to do this without `Array.from()`

Reply:
Thanks.